lib/services/queue_service.dart

```diff
@@ -171,6 +171,44 @@ class QueueService {
   Future<List<jellyfin_models.BaseItemDto>> generateRadioTracks(int numSongs) async {
     List<jellyfin_models.BaseItemDto> itemsOut = [];
     switch (FinampSettingsHelper.finampSettings.radioMode) {
+      case RadioMode.shuffle:
+        // Adds songs in such a manner to simulate "shuffle + repeat all", but with each repeat iteration re-shuffling
+        // the order.
+        // Songs added to the queue manually will throw things off, though!
```

Comment:
ahh, good catch. but I think we can work around this by filtering out any tracks with a different queue source.  
Finamp has a "global" queue source for the entire queue, but also individual sources for each track, exactly to differentiate between tracks added via regular playback or later on via Next Up or some other mechanism. This is what you've been using for adding tracks as coming from "Radio" before.  
So if you filter all `FinampQueueItem`s down to sources where the source ID matches the global queue source ID, it should not be affected by manually added tracks anymore.  
Maybe it introduces new edge cases, but let's see :)